nipype/pipeline/engine/utils.py

```diff
@@ -329,6 +330,8 @@ def _get_valid_pathstr(pathstr):
     Removes:  [][ (){}?:<>#!|"';]
     Replaces: ',' -> '.'
     """
+    if not isinstance(pathstr, (str, bytes)):
+        pathstr = to_str(pathstr)
```

Comment:
@satra, pathstr was converted to string using the ```__str__``` member. However, for lists, tuples, dict, etc, they fall back to ```repr```.  After adding unicode_literals everywere, ```repr('some string')``` is ```u'some string``` with the u. This u will be there after the pathstr is fixed for not valid characters.